File: src/main/java/dev/yoghurt1131/weatherapi/application/service/WeatherApiService.java
package dev.yoghurt1131.weatherapi.application.service;

import dev.yoghurt1131.weatherapi.application.exception.ApiCallException;
import dev.yoghurt1131.weatherapi.domain.CurrentWeather;
import dev.yoghurt1131.weatherapi.domain.output.valueobject.Forecast;

public interface WeatherApiService {

    CurrentWeather getCurrentWeather(String cityName) throws ApiCallException;

    /**
     * 指定された都市の今日の天気を返す
     * @param cityName
     * @throws ApiCallException
     */
    Forecast getTodaysWeather(String cityName) throws ApiCallException;
}

File: src/main/kotlin/dev/yoghurt1131/weatherapi/application/service/WeatherApiService.kt
package dev.yoghurt1131.weatherapi.application.service

import dev.yoghurt1131.weatherapi.domain.entity.City
import dev.yoghurt1131.weatherapi.application.controller.response.CurrentWeather
import dev.yoghurt1131.weatherapi.application.controller.response.Forecast
import dev.yoghurt1131.weatherapi.application.controller.response.RainForecast

interface WeatherApiService {

    fun getCurrentWeather(cityName: String): CurrentWeather

    fun getTodayWeather(city: City): Forecast

    fun getRainForecast(city: City):List<RainForecast>
}
